lib/src/keyboard.dart

```diff
@@ -84,6 +85,64 @@ extension KeyboardInput on WidgetTester {
     await pumpAndSettle();
   }
 
+  /// Simulates the user pressing ENTER in a widget attached to the IME.
+  /// 
+  /// Instead of key events, this method generates a "\n" insertion followed by a TextInputAction.newline.
+  /// 
+  /// The given [finder] must find a [StatefulWidget] whose [State] implements
+  /// [DeltaTextInputClient].
+  ///
+  /// If the [DeltaTextInputClient] currently has selected text, that text is first deleted,
+  /// which is the standard behavior when typing new characters with an existing selection.
+  Future<void> pressEnterWithIme({
+    Finder? finder,
+    GetDeltaTextInputClient? getter,
+  }) async {
+    if (!testTextInput.hasAnyClients) {
+      // There isn't any IME connections.
+      return;
+    }
+
+    await ime.typeText('\n', finder: finder, getter: getter);
+    await pump();
+    await testTextInput.receiveAction(TextInputAction.newline);
+    await pump();
+  }
+
+  /// Simulates the user pressing ENTER on a widget that could be attached to the IME.
+  /// 
+  /// If the ENTER key isn't handled, generates a "\n" insertion followed by a TextInputAction.newline.
+  ///
+  /// The given [finder] must find a [StatefulWidget] whose [State] implements
```

Comment:
Let's try to use doc macros instead of repeating docs. Otherwise they're likely to get out of sync.